### PR TITLE
Clarify what ‘memory type’ is

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Checklist
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### Description
Closes https://github.com/ethereum/solidity/issues/2167, will also be vastly improved when https://github.com/ethereum/solidity/pull/4615/commits is merged, and that reduces some duplication, and has better linking.

### CODE REVIEW

File: docs/contracts.rst [hunk 1]
```diff
@@ -1201,7 +1201,7 @@ contracts (``L.f()`` if ``L`` is the name of the library). Furthermore,
 ``internal`` functions of libraries are visible in all contracts, just as
 if the library were a base contract. Of course, calls to internal functions
 use the internal calling convention, which means that all internal types
-can be passed and memory types will be passed by reference and not copied.
+can be passed and types `stored in memory <types.html#data-location>`_ will be passed by reference and not copied.
```
Comment (axic) [inline docs/contracts.rst]: I think we should use references as part of RST. I think the syntax is something like `:ref:data-location`

Comment (ChrisChinchilla) [reply docs/contracts.rst]: @axic Yeah, this was kind of a shortcut as the other PR mentioned above adds new links anyway, hmm, I'll redo it properly anyway and we can worry about those conflicts when it comes to merging things later.

**Comment by axic** [comment]
Can you please squash the commits? Should merge it then.

**Comment by ChrisChinchilla** [comment]
@axic Done I hope. I suck when it comes to squashing, and lean too much on the GitHub button doing it for you…

**Comment by axic** [comment]
Yeah Github doesn't do it for you.

On the commandline it is simple: `git rebase -i HEAD~n` where `n` is the number of commits you want to review. it will present you with a list of commits and things you can do with them.

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
Merging as spellcheck and docs finished.

**Pull request merged**